### PR TITLE
Delay showing window until app is ready (reduce flicker) #645

### PR DESCRIPTION
**Pull request by secretfork**
This will make window stay hidden until presentable, as discussed in https://github.com/Automattic/simplenote-electron/issues/645
However, i couldn't test it against the current HEAD revision, since it won't compile cleanly on windows.

### CODE REVIEW

**Comment by dmsnell** [comment]
Nice work!

@roundhill I had wondered about the UX implications here - is it ridiculous to think about a splash screen? should we spend the time to figure out why the app takes so long to load?

@secretfork thanks for this effort - by any chance do you have insight into how long it's taking to be ready and what it's doing until then?

**Comment by secretfork** [comment]
@dmsnell now worries. It's actually not that long. Maybe about 0.5 second or so. But it's the small things that give the impression of "flicker". To be totally honest, many people probably wont notice (or care). It's actually quite common that developers fail to adress this issue, and it's also a common issue since many ui frameworks has some kind of hook/callback like 'ready-to-show' to remedy just this.

I don't think you need to think about a splash screen. Now investigating where the delay is is a better idea, however, i don't think 0,5 seconds is very long. Delaying showing the window until it is presentable is what people do. Not just with electron, but with native toolkits too. It's just that sometimes you don't notice the lack of it (that said, if you present the window before it's ready, it might look ok on a fast desktop machine because it's speedy, but it will degrade on slower hardware, so it's *always*a good idea to wait til it's ready).

I did two short mp4s to illustrate the issue:
before - https://www.dropbox.com/s/46zjrc1wjj2rji4/before.mp4?dl=0
after - https://www.dropbox.com/s/ne6g7u8kmuoghln/after.mp4?dl=0

**Comment by roundhill** [comment]
I'm all or this 👍, creates a nicer UX.

File: desktop/app.js [hunk 1]
```diff
@@ -83,6 +84,11 @@ module.exports = function main() {
       // when you should delete the corresponding element.
       mainWindow = null;
     });
+
+    // wait until window is presentable
+    mainWindow.once('ready-to-show', function() {
+        mainWindow.show()
```
Comment (roundhill) [inline desktop/app.js]: Would you mind throwing a semicolon at the end here? Just want to try to match the coding style with the rest of the file.

**Comment by secretfork** [comment]
@roundhill Sorry about the missing semi-colon. Amended.

File: desktop/app.js [hunk 2]
```diff
@@ -83,6 +84,11 @@ module.exports = function main() {
       // when you should delete the corresponding element.
       mainWindow = null;
     });
+
+    // wait until window is presentable
+    mainWindow.once('ready-to-show', function() {
+        mainWindow.show();
+    });
```
Comment (dmsnell) [inline desktop/app.js]: do we need to be passing the `this` context here? if not we can simplify

```js
mainWindow.once('ready-to-show', mainWindow.show);
```

File: desktop/app.js (hunk 1 above)
Comment (dmsnell) [reply desktop/app.js]: On that note a run through of `prettier` should take care of any and all styling issues.

File: desktop/app.js (hunk 2 above)
Comment (secretfork) [reply desktop/app.js]: That would work just as well. Let me know if you want me to update the pull request.

**Comment by roundhill** [comment]
I tested this on macOS, and it does help the launch experience to look/feel better. There's not much of a delay difference with this patch, it feels like the same launch time for any native macOS app.

File: desktop/app.js (hunk 2 above)
Comment (roundhill) [reply desktop/app.js]: @secretfork cool if you don't mind updating to that one last time I will merge this!

Comment (secretfork) [reply desktop/app.js]: @roundhill pushed the change dmsnell suggested (i assume that is what you meant), so feel free to review and merge :+1:

**Comment by roundhill** [comment]
Thanks again @secretfork!

**Pull request merged**